app/models/concerns/episode_media.rb

```diff
@@ -111,39 +192,119 @@ def video_content_type?(*args)
   end
 
   def media_duration
-    media.inject(0.0) { |s, c| s + c.duration.to_f } + podcast.try(:duration_padding).to_f
+    if override?
+      external_media_resource&.duration
+    else
+      media.inject(0.0) { |s, c| s + c.duration.to_f } + podcast.try(:duration_padding).to_f
+    end
   end
 
   def media_file_size
-    media.inject(0) { |s, c| s + c.file_size.to_i }
+    if override?
+      external_media_resource&.file_size
+    else
+      media.inject(0) { |s, c| s + c.file_size.to_i }
+    end
   end
 
+  # 1) must_be_complete or not
+  # that is called with "true" in 2 places
+  # - in checking valid on 1st time publish
+  # - displaying "complete" for status
+  # So really this is about 1st time, must be fully ready,
+  # as there is no prior published media version to fall back on?
+  # And for status display or api responses, are media in the current version done?
+  #
+  # 2) override or not
+  # If there is an override, that is the only status to check
+  # We need a (different?) method to check contents ready for media versions
+  # but if an override is provided, we don't look at uploaded media status (uncut or contents)
+  #
+  # 3) enough segments
+  # Do we have enough media files for the segment count?
+  # That should get checked regardless of must_be_complete
+  #
+  # 4) uncut present
+  # If this is not must_be_complete, then
+  # If a single file upload, is there file uploaded, that isn't marked deleted
+  #
+  # 5) TODO: invalid or error status?
+  # should counts of files include invalid/error status files?
+  # should it be ready if any files is in error or invalid state?
   def media_ready?(must_be_complete = true)
-    if media.empty?
+    if !must_be_complete && medium_uncut?
+      uncut.present? && !uncut.marked_for_destruction?
+    elsif media.empty?
       false
-    elsif must_be_complete && !media.all?(&:status_complete?)
+    elsif segment_count.nil? && (media.size < media.map(&:position).max)
       false
-    elsif segment_count.nil?
-      media.size == media.map(&:position).max
+    elsif media.size < segment_count.to_i
+      false
+    elsif must_be_complete
+      media.all?(&:status_complete?)
     else
-      media.size >= segment_count
+      true
     end
   end
 
   def media_status
-    states = media.map(&:status).uniq
+    states = if override?
+      [external_media_resource&.status]
```

Comment:
So if this is audio/video/uncut... and you overrode the enclosure_url... this will show the external media status, regardless of the contents status.  Probably fine, since these overrides get set programmatically for now.

Reply:
yes, once there is an override url, then that is the url that will be used in the enclosure, so the displayed status needs to reflect that override's status. For the API, the complete/ready media would still be returned, but for the user, the status of the override is all they will get.